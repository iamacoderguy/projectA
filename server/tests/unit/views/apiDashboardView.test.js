const pug = require('pug');
const renderPug = (view, data) => pug.renderFile(view, data);

describe('apiDashboardView', () => {
    it('should render apiDashboardView correctly', () => {
        expect(renderPug('views/apiDashboardView.pug', {
            path: 'path-to-files',
            dashboardName: 'Dashboard',
            ipAddress: '127.3.2.1',
            port: 5000
          })).toMatchSnapshot();
    })
})