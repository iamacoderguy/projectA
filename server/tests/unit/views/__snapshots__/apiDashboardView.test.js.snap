// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apiDashboardView should render apiDashboardView correctly 1`] = `"<!DOCTYPE html><html lang=\\"en-US\\"><head><title>Dashboard</title><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><meta name=\\"description\\" content=\\"Dashboard\\"><link rel=\\"stylesheet\\" href=\\"w3.css\\"></head><body><div class=\\"w3-container\\"><div class=\\"w3-container w3-dark-teal\\"><h2>Dashboard - 127.3.2.1:5000</h2></div><p><iframe width=\\"100%\\" height=\\"200vw\\" title=\\"Result box\\" srcdoc=\\"&lt;i&gt;Result...&lt;/i&gt;\\" name=\\"iframe_result\\"></iframe></p><form class=\\"w3-container\\" method=\\"POST\\" action=\\"/api/auth/connect\\" target=\\"iframe_result\\"><p><label class=\\"w3-text-dark-teal\\"><b>Connect to server (/api/files/auth/connect)</b></label></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">POST</button></p></form><form class=\\"w3-container\\" method=\\"POST\\" action=\\"/api/auth/disconnect\\" target=\\"iframe_result\\"><p><label class=\\"w3-text-dark-teal\\"><b>Disconnect to server (/api/files/auth/disconnect)</b></label></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">POST</button></p></form><form class=\\"w3-container\\" method=\\"GET\\" action=\\"/api/files\\" target=\\"iframe_result\\"><p><label class=\\"w3-text-dark-teal\\"><b>Get the shared files list (/api/files)</b></label></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">GET</button></p></form><form class=\\"w3-container\\" method=\\"GET\\" action=\\"/api/files\\" target=\\"_blank\\"><p><label class=\\"w3-text-dark-teal\\" for=\\"filenameInput\\"><b>Get a shared file (/api/files/:filename)</b></label><input class=\\"w3-input w3-border w3-round\\" id=\\"filenameInput\\" name=\\"filename\\" type=\\"text\\" placeholder=\\"filename\\"></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">GET</button></p></form><form class=\\"w3-container\\" method=\\"POST\\" action=\\"/api/files\\" enctype=\\"multipart/form-data\\" target=\\"iframe_result\\"><p><label class=\\"w3-text-dark-teal\\" for=\\"filepathInput\\"><b>Upload a new file (/api/files)</b></label><input class=\\"w3-input w3-border w3-round\\" id=\\"filepathInput\\" name=\\"file\\" type=\\"file\\"></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">POST</button></p></form></div></body></html>"`;
